File: trunk/src/main/java/net/esper/event/property/PropertyListBuilder.java
package net.esper.event.property;

import net.esper.event.EventPropertyDescriptor;

import java.util.List;

/**
 * Interface for an introspector that generates a list of event property descriptors
 * given a clazz. The clazz could be a JavaBean-style class or any other legacy type.
 */
public interface PropertyListBuilder
{
    /**
     * Introspect the clazz and deterime exposed event properties.
     * @param clazz to introspect
     * @return list of event property descriptors
     */
    public List<EventPropertyDescriptor> assessProperties(Class clazz);
}

File: NEsper/events/property/PropertyListBuilder.cs
using System;
using System.Collections.Generic;

using net.esper.events;

namespace net.esper.events.property
{
	/// <summary>
    /// Interface for an introspector that generates a list of event property descriptors
	/// given a clazz.
	/// </summary>
	
    public interface PropertyListBuilder
	{
		/// <summary>
		/// Introspect the type and deterime exposed event properties.
		/// </summary>
		/// <param name="type">type to introspect</param>
		/// <returns>list of event property descriptors</returns>

        IList<EventPropertyDescriptor> AssessProperties(Type type);
	}
}
